src/plugins/auto_batch/src/compiled_model.cpp

```diff
@@ -154,93 +134,85 @@ std::pair<CompiledModel::WorkerInferRequest&, int> CompiledModel::GetWorkerInfer
             }
         });
     }
-    return {*m_worker_requests.back(), static_cast<int>(batch_id)};
+    return {m_worker_requests.back(), static_cast<int>(batch_id)};
 }
 
-InferenceEngine::IInferRequestInternal::Ptr CompiledModel::CreateInferRequest() {
-    if (!m_model_with_batch) {
-        auto res = m_model_without_batch->CreateInferRequest();
-        res->setPointerToExecutableNetworkInternal(shared_from_this());
-        res->setPointerToSo(m_model_without_batch._so);
-        _so = m_model_without_batch._so;
-        return res;
+std::shared_ptr<ov::IAsyncInferRequest> CompiledModel::create_infer_request() const {
+    if (!m_compiled_model_with_batch) {
+        return m_compiled_model_without_batch->create_infer_request();
     }
-    // trying to create the new API request first
-    InferenceEngine::IInferRequestInternal::Ptr syncRequestImpl = CreateInferRequestImpl(_parameters, _results);
-    if (!syncRequestImpl)
-        syncRequestImpl = CreateInferRequestImpl(_networkInputs, _networkOutputs);
-    syncRequestImpl->setPointerToExecutableNetworkInternal(shared_from_this());
-    InferenceEngine::SoIInferRequestInternal inferRequestWithoutBatch = {m_model_without_batch->CreateInferRequest(),
-                                                                         m_model_without_batch._so};
-    return std::make_shared<AsyncInferRequest>(std::static_pointer_cast<SyncInferRequest>(syncRequestImpl),
-                                               inferRequestWithoutBatch,
-                                               _callbackExecutor);
+
+    auto sync_res = create_sync_infer_request();
+    return std::make_shared<ov::autobatch_plugin::AsyncInferRequest>(
+        std::static_pointer_cast<ov::autobatch_plugin::SyncInferRequest>(sync_res),
+        m_compiled_model_without_batch->create_infer_request(),
+        get_callback_executor());
 }
 
-std::shared_ptr<ngraph::Function> CompiledModel::GetExecGraphInfo() {
-    return m_model_with_batch && m_model_with_batch->GetExecGraphInfo() ? m_model_with_batch->GetExecGraphInfo()
-                                                                        : m_model_without_batch->GetExecGraphInfo();
+std::shared_ptr<const ov::Model> CompiledModel::get_runtime_model() const {
+    return m_compiled_model_with_batch && m_compiled_model_with_batch->get_runtime_model()
```

Comment:
why do we need to check for `m_compiled_model_with_batch->get_runtime_model()` ?

Reply:
follow the original behavior. I think we can remove this check